src/ngraph_builder.cc

```diff
@@ -3870,6 +3870,34 @@ static Status TranslateTransposeOp(
   TF_RETURN_IF_ERROR(
       GetStaticInputVector(op, 1, static_input_map, &permutation));
 
+  auto ng_input_rank = ng_input->get_shape().size();
+  std::unordered_set<int> axes;
+  for (auto p : permutation) {
+    if (p < ng_input_rank && !axes.count(p)) {
+      axes.insert(p);
+    } else {
+      auto missing_axes = 0;
+      for (int i = 0; i < ng_input_rank; i++) {
+        if (std::find(permutation.begin(), permutation.end(), i) ==
+            permutation.end()) {
+          missing_axes = i;
+          break;
+        }
+      }
+
```

Comment:
The current implementation involves an `axes.count(p)` and an `std::find` to search for `i` and `insert`.
I think it can be converted to a counting sort like implementation and getting rid of `axes`. That would avoid the finds we are doing.
```
rank...
vector<bool> count(rank, false);
for (auto p : permutation) {
   if (0 <= p <= rank-1) {
        count[p] = true;
   }
   else {
      //Is this even possible. Maybe TF would raise a error if we pass it stuff outside this range?. 
      //If not raise appropriate error
  }
}
for (i -> 0 to rank){
  if (!count[i]){
    push_back to missing  
 }
}
if (missing.size() > 0){
  raise appropriate error.
}
```

The worst case of this is `2*O(n)`. Though the avg case of the `unordered_set` impl is `O(n)`, the worst case is quadratic because of the finds/counts in the loop. But of course `permutation` is a small array, so ...